habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -261,39 +275,91 @@ def reconfigure(self, config: Config):
                 ao: ao.joint_positions for ao in self.art_objs
             }
 
+    def _setup_targets(self):
+        self._targets = {}
+        for target_handle, transform in self.ep_info["targets"].items():
+            self._targets[target_handle] = mn.Matrix4(
+                [[transform[j][i] for j in range(4)] for i in range(4)]
+            )
+
+    def get_nav_pos(self, pos):
+        pos = mn.Vector3(*pos)
+        height_thresh = 0.15
+        z_min = -0.2
+        use_vs = np.array(self.pathfinder.build_navmesh_vertices())
+
+        if height_thresh is not None:
+            use_vs = use_vs[use_vs[:, 1] < height_thresh]
+        if z_min is not None:
+            use_vs = use_vs[use_vs[:, 2] > z_min]
+        dists = np.linalg.norm(
+            use_vs[:, [0, 2]] - np.array(pos)[[0, 2]], axis=-1
+        )
+
+        closest_idx = np.argmin(dists)
+        return use_vs[closest_idx]
+
     def _recompute_navmesh(self):
-        """Generates the navmesh on the fly. This must be called
+        """Generates the navmesh or loads the saved navmesh if it exists. This must be called
```

Comment:
Perhaps this function should be migrated to the base habitat_sim.py with parameters (e.g. for setting all AOs to STATIC)

Reply:
Yes, I agree this would be a helpful utility to include elsewhere.